test/e2e/e2e_test.go

```diff
@@ -17,32 +17,56 @@ limitations under the License.
 package e2e
 
 import (
+	"flag"
+	"fmt"
+	"path"
 	"testing"
 	"time"
 
-	"github.com/golang/glog"
+	"github.com/onsi/ginkgo"
+	ginkgoconfig "github.com/onsi/ginkgo/config"
+	"github.com/onsi/ginkgo/reporters"
+	"github.com/onsi/gomega"
 	"k8s.io/apimachinery/pkg/util/wait"
-	"k8s.io/client-go/tools/clientcmd"
 
 	"github.com/jetstack/cert-manager/test/e2e/framework"
+	_ "github.com/jetstack/cert-manager/test/e2e/suite"
 )
 
 func init() {
-	framework.RegisterParseFlags()
+	framework.DefaultConfig.AddFlags(flag.CommandLine)
+	flag.Parse()
+
+	// Turn on verbose by default to get spec names
+	ginkgoconfig.DefaultReporterConfig.Verbose = true
+	// Turn on EmitSpecProgress to get spec progress (especially on interrupt)
+	ginkgoconfig.GinkgoConfig.EmitSpecProgress = true
+	// Randomize specs as well as suites
+	ginkgoconfig.GinkgoConfig.RandomizeAllSpecs = true
 
 	wait.ForeverTestTimeout = time.Second * 60
+}
 
-	if "" == framework.TestContext.KubeConfig {
-		glog.Fatalf("environment variable %v must be set", clientcmd.RecommendedConfigPathEnvVar)
+func TestE2E(t *testing.T) {
+	if err := framework.DefaultConfig.Validate(); err != nil {
+		t.Errorf("Invalid test config: %v", err)
+		t.Fail()
 	}
-	if "" == framework.TestContext.CertManagerConfig {
-		glog.Fatalf("environment variable %v must be set", framework.RecommendedConfigPathEnvVar)
+
+	gomega.RegisterFailHandler(ginkgo.Fail)
+
+	// Disable skipped tests unless they are explicitly requested.
+	if ginkgoconfig.GinkgoConfig.FocusString == "" && ginkgoconfig.GinkgoConfig.SkipString == "" {
+		ginkgoconfig.GinkgoConfig.SkipString = `\[Flaky\]|\[Feature:.+\]`
```

Comment:
why skip tests tagged with feature?

Reply:
This was copied over from the old framework, which was originally based on k/k.

👍 should probably get rid of Feature from there. We don't actually use either Flaky or Feature as tags right now anyway.